Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -60,15 +101,97 @@ private void OpenUserInterface(EntityUid user, EntityUid analyzer)
             if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, HealthAnalyzerUiKey.Key, out var ui))
                 return;
 
-            _uiSystem.OpenUi(ui ,actor.PlayerSession);
+            _uiSystem.OpenUi(ui, actor.PlayerSession);
+        }
+
+        /// <summary>
+        /// Mark the entity as having its health analyzed, and link the analyzer to it
+        /// </summary>
+        private void BeginAnalyzingEntity(EntityUid uid, Entity<HealthAnalyzerComponent> healthAnalyzer)
+        {
+            var healthBeingAnalyzedComponent = EnsureComp<HealthBeingAnalyzedComponent>(uid);
+
+            healthBeingAnalyzedComponent.ActiveAnalyzers.Add(healthAnalyzer);
+
+            //Link the health analyzer to the scanned entity
+            healthAnalyzer.Comp.ScannedEntity = uid;
+            _cell.SetPowerCellDrawEnabled(healthAnalyzer.Owner, true);
+
+            UpdateScannedUser(healthAnalyzer.Owner, uid, true);
         }
 
-        public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
+        /// <summary>
+        /// Remove the analyzer from the active list, and remove the component if it has no active analyzers
+        /// </summary>
+        private void StopAnalyzingEntity(EntityUid uid, Entity<HealthAnalyzerComponent> healthAnalyzer)
         {
-            if (!Resolve(uid, ref healthAnalyzer))
+            if (!TryComp<HealthBeingAnalyzedComponent>(uid, out var healthBeingAnalyzedComponent))
                 return;
 
-            if (target == null || !_uiSystem.TryGetUi(uid, HealthAnalyzerUiKey.Key, out var ui))
+            //Remove analyzer from the list
+            healthBeingAnalyzedComponent.ActiveAnalyzers.Remove(healthAnalyzer);
+
+            //If we were the last, remove the component
+            if (healthBeingAnalyzedComponent.ActiveAnalyzers.Count == 0)
+                RemCompDeferred<HealthBeingAnalyzedComponent>(uid);
+
+            //Unlink the analyzer
+            healthAnalyzer.Comp.ScannedEntity = null;
+
+            UpdateScannedUser(healthAnalyzer.Owner, uid, false);
+            _cell.SetPowerCellDrawEnabled(uid, false);
+        }
+
+        public override void Update(float frameTime)
```

Comment:
Normally we put the Update override below the Initialize one for better readability